src/components/[guild]/AddRewardButton/hooks/useSetRoleImageAndNameFromPlatformData.ts

```diff
@@ -3,13 +3,14 @@ import {
   useAddRewardContext,
 } from "components/[guild]/AddRewardContext"
 import usePinata from "hooks/usePinata/usePinata"
-import { useEffect } from "react"
+import { useEffect, useState } from "react"
 import { useFormContext } from "react-hook-form"
 
 const useSetRoleImageAndNameFromPlatformData = (
   platformImage: string,
   platformName: string
 ) => {
+  const [alreadyUploaded, setAlreadyUploaded] = useState(false)
```

Comment:
I'm wondering if this state should be bound to a `platformImage`, `platformName` pair somehow? 🤔 
Not sure, but this way if the same hook call gets new params, we won't upload on the second one? Not sure 😅

Reply:
Oh yep, in my opinion that isn't a big problem actually, since the users can always upload a new role image later, but we can consider reworking this hook later. For now I think I'll merge it and add an issue to Linear about this 😉